File: data/examples/temporary-field.cpp
class Calculator {
private:
    double operand1;
    double operand2;
    char operation;
    
    // 这些字段只在复杂计算时使用
    double tempResult;
    double intermediateValue;
    bool isComplexCalculation;
    std::vector<double> calculationSteps;
    
public:
    double calculate(double a, double b, char op) {
        operand1 = a;
        operand2 = b;
        operation = op;
        
        if (op == '^') {
            // 只有在幂运算时才使用这些临时字段
            isComplexCalculation = true;
            tempResult = 1;
            intermediateValue = a;
            calculationSteps.clear();
            
            for (int i = 0; i < b; i++) {
                tempResult *= intermediateValue;
                calculationSteps.push_back(tempResult);
            }
            return tempResult;
        } else {
            isComplexCalculation = false;
            // 简单运算时这些字段都没有意义
            tempResult = 0;
            intermediateValue = 0;
            calculationSteps.clear();
            
            if (op == '+') return a + b;
            if (op == '-') return a - b;
            if (op == '*') return a * b;
            if (op == '/') return a / b;
            return 0;
        }
    }
};

File: data/examples/temporary-field.java
import java.util.*;

class Calculator {
    private double operand1;
    private double operand2;
    private char operation;
    
    // These fields are only used in complex calculations
    private double tempResult;
    private double intermediateValue;
    private boolean isComplexCalculation;
    private List<Double> calculationSteps;
    
    public Calculator() {
        this.calculationSteps = new ArrayList<>();
    }
    
    public double calculate(double a, double b, char op) {
        operand1 = a;
        operand2 = b;
        operation = op;
        
        if (op == '^') {
            // Only use these temporary fields during power operations
            isComplexCalculation = true;
            tempResult = 1;
            intermediateValue = a;
            calculationSteps.clear();
            
            for (int i = 0; i < b; i++) {
                tempResult *= intermediateValue;
                calculationSteps.add(tempResult);
            }
            return tempResult;
        } else {
            isComplexCalculation = false;
            // These fields are meaningless during simple operations
            tempResult = 0;
            intermediateValue = 0;
            calculationSteps.clear();
            
            switch (op) {
                case '+': return a + b;
                case '-': return a - b;
                case '*': return a * b;
                case '/': return a / b;
                default: return 0;
            }
        }
    }
    
    public List<Double> getCalculationSteps() {
        // This method only makes sense after complex calculations
        return new ArrayList<>(calculationSteps);
    }
    
    public double getTempResult() {
        // This field is only meaningful during complex calculations
        return tempResult;
    }
}
